index.bs

```diff
@@ -691,21 +691,114 @@ An {{MLOperand}} represents an intermediary graph being constructed as a result
+1. Set [=this=]'s {{MLOperand/[[kind]]}} to |kind|.
+1. Let |desc| be the third argument.
+1. If |desc| is not an {{MLOperandDescriptor}}, throw a "{{TypeError}}" {{DOMException}} and abort these steps.
+1. If the [=byte length=] of |desc| is not supported by the implementation, throw a "{{DataError}}" {{DOMException}} and abort these steps.
+1. If the [=check dimensions=] steps given |descr|.{{MLOperandDescriptor/type}} and |desc|.{{MLOperandDescriptor/dimensions}} return `false`, throw a "{{DataError}}" {{DOMException}} and abort these steps.
+1. Set [=this=]'s {{MLOperand/[[type]]}} to |descr|.{{MLOperandDescriptor/type}}.
+1. Set [=this=]'s {{MLOperand/[[dimensions]]}} to |desc|.{{MLOperandDescriptor/dimensions}}.
+1. If |kind| is `"constant"`:
+    1. Let |bufferView| be the next argument.
+    1. If |bufferView| is not an {{MLBufferView}}, throw a "{{DataError}}" {{DOMException}} and abort these steps.
+    1. If [=this=]'s {{MLOperand/[[type]]}} is not equal to |bufferView|'s [=element type=], throw a "{{DataError}}" {{DOMException}} and abort these steps.
+    1. If the [=byte length=] of |desc| is not supported by the implementation, throw a "{{DataError}}" {{DOMException}} and abort these steps.
+    1. If the [=byte length=] of |desc| is not equal with |bufferView|.\[[ByteLength]], throw a "{{DataError}}" {{DOMException}} and abort these steps.
+    1. Set [=this=]'s {{MLOperand/[[tensorData]]}} to |bufferView|.
+    1. Make a request to the underlying platform to register [=this=] as a constant.
+        1. If that fails, throw a "{{TypeError}}" {{DOMException}} and abort these steps.
+1. Otherwise, if |kind| is `"input"`:
+    1. Let |name| be the next argument.
+    1. If |name| is not a [=string=], throw a "{{TypeError}}" {{DOMException}} and abort these steps.
+    1. Set [=this=]'s {{MLOperand/[[name]]}} to |name|.
+    1. Make a request to the underlying platform to register [=this=] as an input.
+        1. If that fails, throw a "{{TypeError}}" {{DOMException}} and abort these steps.
+1. Otherwise, of |kind| is `"output"`:
+    1. Make a request to the underlying platform to register [=this=] as an output.
+        1. If that fails, throw a "{{TypeError}}" {{DOMException}} and abort these steps.
+1. Return [=this=].
+
+To <dfn>check dimensions</dfn> given |dimensions| and |type|, run the following steps:
+1. If |dimensions| is not an array of positive numbers, return `false`;
+1. If |dimensions|.length is 0, return `false`.
+1. If |dimensions|.length is too large to be supported by the implementation, return `false`.
+1. If any element of |dimensions| is too large to be supported by the implementation, return `false`.
+1. Return `true`.
 
 ### The MLActivation interface ### {#api-mlactivation}
 
 Objects implementing the {{MLActivation}} interface represent activation function types.
 
 <script type=idl>
+typedef object MLActivationOptions;
```

Comment:
Yes, it's the union of all supported options.
Should we explicitly enumerate all of them in a typedef (that would cover a gap in the Web IDL), or should we use a generic dictionary and specify in prose the current meaning? (that would not change the Web IDL, but would clarify the prose).
 I do see value in both approaches.
@anssiko do you have a preference?

Reply:
> The range should be [1, ULONG_MAX]

I think that may be implementation dependent. What we can explicitly check is that it's a positive ulong > 1, but the implementation may have other constraints that it should be able to tell the API user.

Reply:
@zolkis in general if we can define something in Web IDL we should prefer that way over prose. Web IDL is machine-readable and allows for bindings code generation and testing.

Reply:
SGTM. It's a bit unusual to have such long unions, currently 12 of them, but we can easily revert any time, in the case an external reviewer finds that awkward.

Reply:
Filed #363 to track this, if we want to.
For now I will go with the explicit union then.

Reply:
Thanks Dom for that reminder. Looks like we are going with option 1, i.e. generic dictionary + specific spec prose in algorithms.

Reply:
Why do we need to define a new type called `MLActivationOptions`? An `MLActivation` is already an abstract interface that can represent all available activation types (that are available for operator fusions) e.g. the *clamp* activation as an `MLActivation` interface is already created wtih the specific `MLClampOptions` accordingly. 

Also a union of all relevant activation options is unnecessary since the use case for each will only always require just one concrete activation option type e.g. `MLClampOptions` for clamp activation, etc.

Reply:
Chai, are you saying that instead of a union of `MLActivationOptions` we should define a separate type for each respective activation option?

I sympathize with that, and removing the generic object typedef - it doesn't add much value indeed. It is like specifying function interface with generic object parameters.

Reply:
Removed MLActivationOptions.